pkg/cmd/korectl/requestor.go

```
@@ -573,17 +573,20 @@ func (o *OIDC) HasExpiredToken() bool {
 func (c *Requestor) WithConfig(config *Config) *Requestor {
 	auth := config.GetCurrentAuthInfo()
 
-	if auth.OIDC != nil {
+	if auth.OIDC != nil && auth.OIDC.IDToken != "" {
```

Isn't this whole problem because we ignore the error in `HasExpiredToken()`? 

See: `claims, _ := utils.NewClaimsFromRawToken(o.IDToken)`

My problem with this fix is that it's not clear to me where is the issue and how is this fixing it.

How can the `IDToken` be empty? How can the user fix it if that happens

Shouldn't we do the refresh before we do the request? It seems to me a better place and also we can handle the errors, instead of saving it into c.configError